src/test/groovy/graphql/schema/DataFetcherFactoriesTest.groovy

```diff
@@ -0,0 +1,40 @@
+package graphql.schema
+
+import spock.lang.Specification
+
+import java.util.concurrent.CompletableFuture
+import java.util.function.BiFunction
+
+class DataFetcherFactoriesTest extends Specification {
+
+    def cfDF = new StaticDataFetcher(CompletableFuture.completedFuture("hello"))
+    def pojoDF = new StaticDataFetcher("goodbye")
+
+    def "delegation happens as expected"() {
+        given:
+
+        def mapper = new BiFunction<DataFetchingEnvironment, Object, Object>() {
+            @Override
+            Object apply(DataFetchingEnvironment dataFetchingEnvironment, Object o) {
+                return String.valueOf(o) + " world"
+            }
+        }
+        when:
+        def cfWrappedDF = DataFetcherFactories.wrapDataFetcher(cfDF, mapper)
+        def pojoWrappedDF = DataFetcherFactories.wrapDataFetcher(pojoDF, mapper)
+
+        then:
+        (cfWrappedDF.get(null) as CompletableFuture).join() == "hello world"
```

Comment:
just in general: I think it is better never to call `join` but to wait for for the result only a limited amount of time. This way we don't have blocking tests (without any real timeout) if it fails. (We have more tests with a blocking `join` or similar stuff, so it is not a big deal, but we should maybe consider in the future)

Reply:
Yeah I mostly agree however in this unit test I KNOW that the CF is already completed.  By design I know its finished and not async

Reply:
Point taken though